Assets/Scripts/PlayerCharacter.cs

```diff
@@ -0,0 +1,256 @@
+        {
+            jumpNumber = 0; //if the character is not moving vertically, the player has taken at least a moment to rest before jumping again.
+        }
+        if (jumpNumber == 0)
+        {
+            speedJumpReducer = 1; //first jump has no vertical distance reducer.
+        }
+        UpdateAnimationParameters();
+
+    }
+
+    private void UpdateAnimationParameters()
+    {
+        grounded = Physics2D.OverlapCircle(groundCheck.position, groundRadius, whatIsGround);
+        anim.SetBool("IsOnGround", grounded);
+        //sets tells animator if character is falling or on the ground.
+        if (myRigidBody.velocity.y < 0 && grounded == false)
+        {
+            falling = true;
+        }
+        else if (grounded==true || myRigidBody.velocity.y <= 0)
+        {
+            falling = false;
+        }
+        anim.SetBool("Falling", falling);
+        anim.SetFloat("Speed", Math.Abs(myRigidBody.velocity.x)); //lets the animator know the speed of player.
+    }
+
+    private void FixedUpdate()
+    {
+        UpdatePhysicsMaterial();
+        if (Time.time - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
+        {
+            //character cannot use the move function when dying.
+            if (Input.GetKey(KeyCode.D) || Input.GetKey(KeyCode.A))
+            {
+                Move();//lets player move a fixed amount not dependent on frames
+            }
+            //player flips sprite when going opposite direction of prior movement.
+            if (direction > 0 && !facingRight)
+            {
+                Flip();
+            }
+            else if (direction < 0 && facingRight)
+            {
+                Flip();
+            }
+            deathTimer = 0;
+        }
+    }
+
+    private void UpdatePhysicsMaterial()
+    {
+        //changes physics material when moving or stopping.
+        if (Mathf.Abs(direction) > 0)
+        {
+            playerGroundCollider.sharedMaterial = playerMovingPhysicsMaterial;
+        }
+        else
+        {
+            playerGroundCollider.sharedMaterial = playerStoppingPhysicsMaterial;
+        }
+    }
+
+    private void GetMovementInput()
+    {
+        //sets direction of movement from player commands
+        direction = Input.GetAxisRaw("Horizontal");
+    }
+
+    private void Move()
+    {
+        myRigidBody.AddForce(Vector2.right * direction * acceleration / speedJumpReducer);
+        Vector2 clampedVelocity = myRigidBody.velocity;
+        clampedVelocity.x = Mathf.Clamp(myRigidBody.velocity.x, -maxSpeed / speedJumpReducer, maxSpeed / speedJumpReducer);
+        /*if (isInDeath)
+        {
+            clampedVelocity.x = clampedVelocity.x/10;
+        }*/
+        myRigidBody.velocity = clampedVelocity;//adds up a build up of acceleration to a maximum velocity
+    }
+
+    private void Jump()
+    {
+        //Allows player to jump twice before having to fall to the ground.
+       if (jumpNumber == 0&&grounded==true)
```

Comment:
add spaces between values and operators, don't use `== true` just state the name of the bool (it resolves to true or false) (e.g. `jumpNumber == 0 && grounded)